getting-started/Frontend/emojify/src/main/java/com/google/cloud/kotlin/emojify/ImageActivity.kt

```diff
@@ -0,0 +1,243 @@
+ */
+
+package com.google.cloud.kotlin.emojify
+
+import android.graphics.Color
+import android.net.Uri
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.GridLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.util.Log
+import android.util.NoSuchPropertyException
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.Toast
+import com.android.volley.DefaultRetryPolicy
+import com.android.volley.Request
+import com.android.volley.Response
+import com.android.volley.toolbox.JsonObjectRequest
+import com.android.volley.toolbox.Volley
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.engine.DiskCacheStrategy
+import com.bumptech.glide.request.RequestOptions
+import com.bumptech.glide.request.target.Target
+import com.bumptech.glide.signature.ObjectKey
+import com.yanzhenjie.album.Album
+import com.yanzhenjie.album.AlbumFile
+import com.yanzhenjie.album.api.widget.Widget
+import com.yanzhenjie.album.impl.OnItemClickListener
+import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.storage.StorageMetadata
+import com.google.firebase.storage.StorageReference
+import com.yanzhenjie.album.widget.divider.Api21ItemDivider
+import kotlinx.android.synthetic.main.activity_list_content.image_view
+import kotlinx.android.synthetic.main.activity_list_content.tv_message
+import kotlinx.android.synthetic.main.activity_list_content.recycler_view
+import kotlinx.coroutines.experimental.CommonPool
+import kotlinx.coroutines.experimental.Job
+import kotlinx.coroutines.experimental.launch
+import kotlinx.android.synthetic.main.toolbar.toolbar
+import java.io.File
+import java.util.ArrayList
+import java.util.Properties
+
+class ImageActivity : AppCompatActivity() {
+
+    private lateinit var backendUrl: String
+    private var adapter: Adapter? = null
+    private var albumFiles: ArrayList<AlbumFile>? = null
+    private val job = Job()
+
+    companion object {
+        private var emjojifiedUrl: String = ""
+        private var imageId: String = ""
+        private val storageRef: StorageReference = FirebaseStorage.getInstance().reference
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        show("First, select picture to emojify!")
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_album)
+        setSupportActionBar(toolbar)
+        recycler_view.layoutManager = GridLayoutManager(this, 3) as RecyclerView.LayoutManager
+        val divider = Api21ItemDivider(Color.TRANSPARENT, 10, 10)
+        recycler_view.addItemDecoration(divider)
+        adapter = Adapter(this, OnItemClickListener { _, position -> previewImage(position) })
+        recycler_view.adapter = adapter
+
+        val properties = Properties()
+        properties.load(assets.open("application.properties"))
+        if (properties["storage.bucket.name"] == null)
```

Comment:
I think you can do
`val bucketName = properties["storage.bucket.name"]?: throw NoSuchPropertyException("property 'storage.bucket.name' doesn't exist in application.properties!")`

See: https://kotlinlang.org/docs/reference/null-safety.html